03-raycast.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019 LG Electronics, Inc.
+#
+# This software contains code licensed as described in LICENSE.
+#
+
+import simulation.config as config
+import lgsvl
+
+cf = config.Config()
+sim = cf.Simulator()
+
+cf.LoadOrResetScene(sim, "BorregasAve")
+
+# The next few lines spawns an EGO vehicle in the map
+spawns = sim.get_spawn()
+
+state = lgsvl.AgentState()
+state.transform = spawns[0]
+forward = lgsvl.utils.transform_to_forward(state.transform)
+right = lgsvl.utils.transform_to_right(state.transform)
+up = lgsvl.utils.transform_to_up(state.transform)
+sim.add_agent("Lincoln2017MKZ (Apollo 5.0)", lgsvl.AgentType.EGO, state)
+
+# This is the point from which the rays will originate from. It is raised 1m from the ground
+p = spawns[0].position
+p.y += 1
+
+# useful bits in layer mask
+# 0 - Default (road & ground)
+# 9 - EGO vehicles
+# 10 - NPC vehicles
+# 11 - Pedestrian
+# 12 - Obstacle
+
+# Included layers can be hit by the rays. Otherwise the ray will go through the layer
+layer_mask = cf.Raycast_Layer()
+
+# raycast returns None if the ray doesn't collide with anything
+# hit also has the point property which is the Unity position vector of where the ray collided with something
+hit = sim.raycast(p, right, layer_mask)
+if hit:
+  print("Distance right:", hit.distance)
+
+hit = sim.raycast(p, -right, layer_mask)
+if hit:
+  print("Distance left:", hit.distance)
```

Comment:
since layermask can show what object it hits based on the return bits, shall we show that msg on the printed output?

Reply:
Most of the time, I do not think it is necessary. But we can add this function to the backlog.